src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -1914,6 +1914,7 @@ public boolean addTransaction(SIPTransaction transaction) {
                 && !firstTransactionId.equals(transaction.getBranchId())
                 && transaction.getMethod().equals(firstTransactionMethod)) {
             setReInviteFlag(true);
+            ackProcessed = false;
```

Comment:
have you tested this change somehow? would it be possible to add a junit test that test this condition?